lang/src/org/partiql/lang/eval/physical/operators/FilterRelationalOperatorFactory.kt

```diff
@@ -1,21 +1,69 @@
 package org.partiql.lang.eval.physical.operators
 
 import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.eval.booleanValue
+import org.partiql.lang.eval.isNotUnknown
+import org.partiql.lang.eval.physical.EvaluatorState
+import org.partiql.lang.eval.relation.RelationIterator
+import org.partiql.lang.eval.relation.RelationType
+import org.partiql.lang.eval.relation.relation
+import org.partiql.lang.planner.transforms.DEFAULT_IMPL_NAME
 
-/** Provides an implementation of the [PartiqlPhysical.Bexpr.Filter] operator.*/
+/**
+ * Provides an implementation of the [PartiqlPhysical.Bexpr.Filter] operator.
+ *
+ * @constructor
+ *
+ * @param name
+ */
 abstract class FilterRelationalOperatorFactory(name: String) : RelationalOperatorFactory {
-    final override val key: RelationalOperatorFactoryKey = RelationalOperatorFactoryKey(RelationalOperatorKind.FILTER, name)
 
-    /** Creates a [RelationExpression] instance for [PartiqlPhysical.Bexpr.Filter]. */
+    final override val key = RelationalOperatorFactoryKey(RelationalOperatorKind.FILTER, name)
+
+    /**
+     * Creates a [RelationExpression] instance for [PartiqlPhysical.Bexpr.Filter].
+     *
+     * @param impl
+     * @param predicate
+     * @param sourceBexpr
+     * @return
+     */
     abstract fun create(
-        /**
-         * Contains any static arguments needed by the operator implementation that were supplied by the planner
-         * pass which specified the operator implementation.
-         */
         impl: PartiqlPhysical.Impl,
-        /** Invokes the filter's predicate. */
         predicate: ValueExpression,
-        /** Invokes the source binding expression. */
         sourceBexpr: RelationExpression
     ): RelationExpression
 }
+
+internal object FilterRelationalOperatorFactoryDefault : FilterRelationalOperatorFactory(DEFAULT_IMPL_NAME) {
+    override fun create(
+        impl: PartiqlPhysical.Impl,
+        predicate: ValueExpression,
+        sourceBexpr: RelationExpression
+    ) = SelectOperatorDefault(
+        input = sourceBexpr,
+        predicate = predicate
+    )
+}
+
+internal class SelectOperatorDefault(
```

Comment:
Is there a reason why this is named as SelectOperator instead of FilterOperator?

Reply:
https://en.wikipedia.org/wiki/Relational_algebra#Selection_(%CF%83)

Reply:
Doesn't matter though since it's internal. Calcite uses the term Filter for Selection; probably to disambiguate from the projection SELECT clause